arelle/plugin/validate/NL/rules/fg_nl.py

```diff
@@ -193,3 +194,43 @@ def rule_fg_nl_09(
             msg=_('A fact SHOULD NOT contain an "id" attribute'),
             modelObject=facts,
         )
+
+
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[
+        DISCLOSURE_SYSTEM_NT16,
+        DISCLOSURE_SYSTEM_NT17,
+        DISCLOSURE_SYSTEM_NT18,
+    ],
+)
+def rule_fg_nl_11(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation] | None:
+    """
+    FG-NL-11: XBRL instance file Should only contain standard units based on the XBRL units Registry.
+    """
+    modelXbrl = val.modelXbrl
+    customUnits = []
+    utrItemTypeEntries = ValidateUtr(modelXbrl).utrItemTypeEntries
+    validUtrUnits = set()
+    for entries in utrItemTypeEntries.values():
+        for entry in entries.values():
+            if entry.nsUnit and entry.unitId:
+                validUtrUnits.add((entry.nsUnit, entry.unitId))
+    for unitId, unit in modelXbrl.units.items():
+        for measures in unit.measures:
+            for measure in measures:
+                if (measure.namespaceURI, measure.localName) not in validUtrUnits:
+                    customUnits.append((unit, measure))
```

Comment:
nit: instead of collecting custom units you could yield a validation warning here